common/types/map.go

```diff
@@ -125,17 +145,27 @@ func (m *baseMap) Equal(other ref.Value) ref.Value {
 	it := m.Iterator()
 	for it.HasNext() == True {
 		key := it.Next()
-		if otherVal := otherMap.Get(key); IsError(otherVal.Type()) {
+		if otherVal := otherMap.Get(key); IsError(otherVal) {
 			return False
```

Comment:
Since there are two reasons why the `Get` could fail, I'm beginning to understand why you might want to consider this an error. Value conversion errors are somewhat uncommon, but still possible through this path. The more common error is the `no such key` error which indicates absence of the key from the map.

I'll add a TODO to refine this behavior. I had played around with alternative representations which might help disambiguate these two error conditions, but it's not ready for incorporation here.